modernizer-annotation-processor/src/main/java/org/gaul/modernizer_annotation_processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,341 @@
+                makeFile(new File(outputDir,
+                    ModernizerAnnotationUtils.IGNORE_METHODS_FILE_NAME),
+                    annotatedElements.getAnnotatedMethods());
+            }
+        }
+        return true;
+    }
+
+    private AnnotatedElements getAnnotatedElements(
+        RoundEnvironment roundEnv,
+        TypeElement annotation
+    ) {
+        List<String> annotatedClasses = new ArrayList<String>();
+        List<String> annotatedMethods = new ArrayList<String>();
+        for (Element element : roundEnv.getElementsAnnotatedWith(annotation)) {
+            List<String> warnings = Arrays.asList(
+                    element.getAnnotation(SuppressWarnings.class).value());
+            if (warnings.contains("modernizer")) {
+                if (element.getKind().isClass()) {
+                    annotatedClasses.add(
+                        getFullClassNameRegex(element));
+                } else if (element.getKind().equals(ElementKind.METHOD) ||
+                    element.getKind().equals(ElementKind.CONSTRUCTOR)) {
+                    annotatedMethods.add(getMethodRepresentation(element));
+                }
+            }
+        }
+        return new AnnotatedElements(annotatedClasses, annotatedMethods);
+    }
+
+
+    private void makeFile(
+        File file,
+        List<String> annotatedElements
+    ) {
+        Writer writer = null;
+        try {
+            writer = new BufferedWriter(new FileWriter(file));
+            for (String element : annotatedElements) {
+                writer.write(element + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (writer != null) {
+                try {
+                    writer.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+
+    private File getOutputDirectory() {
+        try {
+            FileObject fileObjectToGetPath =
+                processingEnv.getFiler().createResource(
+                    StandardLocation.CLASS_OUTPUT,
+                    "",
+                    "dummy-file.txt");
+            File outputDir =
+                new File(fileObjectToGetPath.getName()).getParentFile();
+            return ModernizerAnnotationUtils.getOutputDir(outputDir);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Returns a formatted version of the fully qualified class name
+     * regular expression of the element to perform the necessary
+     * checks in the plugin against the ASM parsed class name.
+     * @param classElement The element of kind ElementKind.CLASS whose
+     * formatted fully qualified class name regex is to be returned
+     * @return Formatted fully qualified class name regex of the given element
```

Comment:
This description is redundant. I'd just leave the overall description and let these be empty